Formula/erlang.rb

```diff
@@ -32,6 +33,7 @@ class Erlang < Formula
   depends_on "libtool" => :build
   depends_on "openssl"
   depends_on "fop" => :optional # enables building PDF docs
+  depends_on :java => :optional
```

Comment:
There's a mismatch here, you've called the option `with-javac` but the dependency is `java`. One won't trigger the other.